lib/pages/authentication/forms/MiscForm.dart

```diff
@@ -373,4 +365,25 @@ class MiscFormState extends State<MiscForm> {
 
     widget.formData.hobbies = formHobbies;
   }
+
+
+  void saveData() {
+    getRoommate();
+    getDistance();
+    getHobbies();
+    widget.formData.semester = int.parse(currentSemester);
+    widget.formData.gradesPath = gradesKey.currentState.path;
+  }
+
+  void loadNext() {
+    DataFetcher.fetchFormData(widget.formData).then((success) {
+      if (success) {
+        Router.pop(context);
+      } else {
+        showSnackBar("Something went wrong!");
+      }
+    });
+    Router.replaceAll(context, '/home');
```

Comment:
This seems wrong. If the request isn't successful we can't just close everything and redirect to `/home`.